backend/controller/evaljob_controller.go

```diff
@@ -0,0 +1,568 @@
+/*
+Copyright 2024.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controller
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/tools/record"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	lmevalservicev1beta1 "github.com/foundation-model-stack/fms-lm-eval-service/api/v1beta1"
+	"github.com/go-logr/logr"
+)
+
+const (
+	DriverPath                  = "/bin/driver"
+	DestDriverPath              = "/opt/app-root/src/bin/driver"
+	PodImageKey                 = "pod-image"
+	DriverImageKey              = "driver-image"
+	DriverServiceAccountKey     = "driver-serviceaccount"
+	DefaultPodImage             = "quay.io/yhwang/lm-eval-aas-flask:test"
+	DefaultDriverImage          = "quay.io/yhwang/lm-eval-aas-driver:test"
+	DefaultDriverServiceAccount = "driver"
+)
+
+// EvalJobReconciler reconciles a EvalJob object
+type EvalJobReconciler struct {
+	client.Client
+	Scheme    *runtime.Scheme
+	Recorder  record.EventRecorder
+	ConfigMap string
```

Comment:
How does the user point to this `ConfigMap`? I'm not seeing it in the CRD (could be missing it somewhere).

Reply:
I plan to use the `ConfigMap` to store settings for the controller and the fields in the REST APIs. For the sake of the argument, if there is a `size` property in the REST API that allows users to specify the pod's spec. we may have a `big-size` settings in the `ConfigMap` and it contains the 4000m CPU and 8GB memory settings. and a median and small size settings. So it's not directly used by the CR, but somehow related.